docs/proposals/deploy-csi-to-k3s.md

```diff
@@ -0,0 +1,61 @@
+# Proposal:  K3s Kubernetes distribution 
+
+Last updated: 12.04.22
+
+
+## Abstract
+
+Specify required changes to support working CSI on k3s, describe steps for manual patching scheduler 
+
+## Background
+
+Currently csi is compatible with Vanilla Kubernetes, RKE2, OpenShift and Kind. However, Kind is only used for testing purposes. We want to support new platform.  
+
+## Proposal
+
+Need to create next config file in directory `/var/lib/rancher/k3s/agent/pod-manifests`:
+```
+apiVersion: kubescheduler.config.k8s.io/v1beta1
+kind: KubeSchedulerConfiguration
+extenders:
+  - urlPrefix: "http://127.0.0.1:8889"
+    filterVerb: filter
+    prioritizeVerb: prioritize
+    weight: 1
+    enableHTTPS: false
+    nodeCacheCapable: false
+    ignorable: true
+    httpTimeout: 15s
+leaderElection:
+  leaderElect: true
+clientConnection:
+  kubeconfig: /var/lib/rancher/k3s/server/cred/scheduler.kubeconfig
+```
+After that there are two ways:
+1. If you haven't started the service yet, run service creation command with the following flag `k3s server --kube-scheduler-arg=config=/var/lib/rancher/k3s/agent/pod-manifests/scheduler/config.yaml`
+2. If you already have a running service, modify  `/etc/systemd/system/k3s.service`. Add at the end of the file `--kube-scheduler-arg=config=/var/lib/rancher/k3s/agent/pod-manifests/scheduler/config.yaml` and manually run the command for reloading daemon `systemctl daemon-reload && systemctl restart k3s`.
```

Comment:
I guess we should tell that we choose the second one for automation purpose (if it's correct)